pkg/controller/atlasdeployment/atlasdeployment_controller.go

```diff
@@ -276,6 +276,10 @@ func (r *AtlasDeploymentReconciler) handleAdvancedDeployment(ctx *workflow.Conte
 		return result, nil
 	}
 
+	if searchResult := ensureAtlasSearch(context.Background(), ctx, project.ID(), deployment); !searchResult.IsOk() {
```

Comment:
I assume this func can be moved directly to the `Reconcile()` to avoid duplication.
Similar to `handleAdvancedOptions()`